### PR TITLE
feat: add translations support

### PR DESCRIPTION
**Pull request by BryanttV**
### Description
This PR adds translation support to the plugin. It only adds the translation to **Spanish (Latin America)** and **Spanish (Spain)**.

### How to Test
1. Install this plugin with the changes in this branch in your environment.
2. Change the platform language in (**Account** > **Site Preferences** > **Site Language**). Please set it to **Spanish (Latin America)** or **Spanish (Spain)**.
3. In **LMS** > **Instructor** > **OnTask** you can see the template with the corresponding translations.

### Screenshots
image
image

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -87,13 +92,14 @@ selfcheck: ## check that the Makefile is well-formed
...
 ## Localization targets
 
-extract_translations: ## extract strings to be translated, outputting .mo files
-	rm -rf docs/_build
-	cd platform_plugin_ontask && ../manage.py makemessages -l en -v1 -d django
-	cd platform_plugin_ontask && ../manage.py makemessages -l en -v1 -d djangojs
+extract_translations: ## extract strings to be translated, outputting .po files
+	cd $(PACKAGE) && i18n_tool extract
+	mv $(EXTRACTED_DJANGO) $(EXTRACTED_TEXT)
+	if [ -f "$(EXTRACTED_DJANGOJS)" ]; then cat $(EXTRACTED_DJANGOJS) >> $(EXTRACTED_TEXT); rm $(EXTRACTED_DJANGOJS); fi
```
Comment (mariajgrimaldi) [inline Makefile]: Is there a reason why changing makemessages for i18n_tool?

File: Makefile [hunk 2]
```diff
@@ -87,13 +92,14 @@ selfcheck: ## check that the Makefile is well-formed
...
 
-compile_translations: ## compile translation files, outputting .po files for each supported language
-	cd platform_plugin_ontask && ../manage.py compilemessages
+compile_translations: ## compile translation files, outputting .mo files for each supported language
+	cd $(PACKAGE) && i18n_tool generate -v
+	python manage.py compilejsi18n --namespace OnTaskPluginI18n --output $(JS_TARGET)
```
Comment (mariajgrimaldi) [inline Makefile]: Same question here but about compilemessages

File: Makefile (hunk 1 above)
Comment (BryanttV) [reply Makefile]: The change was only to replicate the way it is done in [XBlocks](https://github.com/eduNEXT/xblock-completion-grading/blob/4c4877c1ce8a54d542a9e85c025f6e9d9e225ccd/Makefile#L99-L106)

File: Makefile (hunk 2 above)
Comment (BryanttV) [reply Makefile]: Same above

File: Makefile (hunk 1 above)
Comment (mariajgrimaldi) [reply Makefile]: do we need it though?

Comment (mariajgrimaldi) [inline Makefile]: I'd prefer if we keep this as is, and then figure out the best way of adding translations in a plugin like this. But let's follow the standard we've used so far. What do you think?

File: platform_plugin_ontask/public/js/translations/en/text.js [hunk 3]
```diff
@@ -0,0 +1,142 @@
+
```
Comment (mariajgrimaldi) [inline platform_plugin_ontask/public/js/translations/en/text.js]: do we need this?

**Review by mariajgrimaldi — COMMENTED** [review]
is this PR ready for review?

File: Makefile (hunk 1 above)
Comment (BryanttV) [reply Makefile]: We could leave it as it is now in the [cookiecutter]( https://github.com/openedx/edx-cookiecutters/blob/a6a80787b102d9811aef4432bd3b709603cf720f/cookiecutter-django-app/%7B%7Bcookiecutter.repo_name%7D%7D/Makefile#L86-L91), what do you think?

File: platform_plugin_ontask/public/js/translations/en/text.js (hunk 3 above)
Comment (BryanttV) [reply platform_plugin_ontask/public/js/translations/en/text.js]: We are not using it, I removed it: https://github.com/eduNEXT/platform-plugin-ontask/commit/27aad67434374b7e16232657482d5185c87dd7f6

**Comment by BryanttV** [comment]
Hi @mariajgrimaldi, this PR is ready for review.

File: Makefile (hunk 1 above)
Comment (mariajgrimaldi) [reply Makefile]: Agreed!

File: platform_plugin_ontask/conf/locale/es_419/LC_MESSAGES/django.po [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+"The OnTask Workflow ID is not set for this course. Please set it in the "
+"Advanced Settings of the course."
+msgstr ""
+"El ID del workflow de OnTask no está configurado para este curso. "
+"Por favor, configúrelo en la Configuración Avanzada del curso."
+
+#: static/html/ontask.html:7
+msgid ""
+"Load the data from the course into OnTask. This will create a new table in "
+"OnTask with the data summary configured in the plugin settings. The default "
```
Comment (mariajgrimaldi) [inline platform_plugin_ontask/conf/locale/es_419/LC_MESSAGES/django.po]: I thought we removed the "in the plugin settings" part

**Review by mariajgrimaldi — APPROVED** [review]

File: platform_plugin_ontask/conf/locale/es_419/LC_MESSAGES/django.po (hunk 4 above)
Comment (BryanttV) [reply platform_plugin_ontask/conf/locale/es_419/LC_MESSAGES/django.po]: I thought so too hahaha, I corrected it: https://github.com/eduNEXT/platform-plugin-ontask/commit/2a796aa78e18102e28c2858ea868f63fbe60e7cf

**Pull request merged**